05.ls/ls.rb

```diff
@@ -83,29 +74,30 @@ def convert_file_stat_chars(file_stat)
 end
 
 def make_file_stats
-  files = fetch_files
+  files, ls_dir = fetch_files
   files.map do |file|
-    file_stat = File.stat(file)
-    file_stats = {}
-    file_stats[:blocks_size] = file_stat.blocks
-    file_stats[:permission] = convert_file_stat_chars(file_stat)
-    file_stats[:hard_link] = file_stat.nlink.to_s
-    file_stats[:user] = Etc.getpwuid(file_stat.uid).name
-    file_stats[:group] = Etc.getgrgid(file_stat.gid).name
-    file_stats[:file_size] = file_stat.size.to_s
-    file_stats[:time] = file_stat.mtime.strftime(' %b %e %R ')
-    file_stats[:file_name] = File.basename(file)
-    file_stats
+    Dir.chdir(ls_dir) do
+      file_stat = File.stat(file)
+      file_stats = {}
+      file_stats[:blocks_size] = file_stat.blocks
+      file_stats[:permission] = convert_file_stat_chars(file_stat)
+      file_stats[:hard_link] = file_stat.nlink.to_s
+      file_stats[:user] = Etc.getpwuid(file_stat.uid).name
+      file_stats[:group] = Etc.getgrgid(file_stat.gid).name
+      file_stats[:file_size] = file_stat.size.to_s
```

Comment:
hard_linkやfile_sizeは数値なので、

```suggestion
      file_stats[:file_size] = file_stat.size
```

のように数値のまま保持した方が再利用性が高くなります。
（`to_s`したものを62行目で再び`to_i`で数値に戻したりするのは無駄な型変換です）